### PR TITLE
Problem: publishing `emerald-cli` via `cargo install` not possbile

### PR DESCRIPTION
**Pull request by sjmackenzie**
Problem: in order to allow distribution of an executable `emerald-cli`
via `cargo install` a separate crate needs to be created focusing only
on CLI functionality (with a dependency on emerald-core).

Secondly the core emerald library needs to expose the FFI in a shared
object and rust crate pub functions for crates.io. This allows users to
import only the core functionality into their apps without having to
bother with the cli functionality.

Solution: split out emerald into two crates `emerald-core` and
`emerald-cli`

Note the CLI functionaly isn't completed this is just to get this
architectural change done so that development can proceed without
throwing mutex locks over the codebase.

### CODE REVIEW

File: emerald-cli/src/main.rs [hunk 1]
```diff
@@ -10,12 +10,12 @@ extern crate log;
 
 extern crate docopt;
 extern crate env_logger;
-extern crate emerald;
+extern crate emerald_core;
```
Comment (sorpaas) [inline emerald-cli/src/main.rs]: I do argue that `extern crate emerald_core as emerald` or something like that would make the CLI code much clearer. But that's up to you.

Comment (sjmackenzie) [inline emerald-cli/src/main.rs]: sure this can be changed in another PR
